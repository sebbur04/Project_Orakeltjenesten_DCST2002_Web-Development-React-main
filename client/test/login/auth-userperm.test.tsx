// Import neccessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import UserPermissions from "../../src/login/userperm";
import userService from "../../src/service/user-service";
import permissionService from "../../src/service/permission-service";

jest.mock("../../src/service/user-service", () => ({
  getAuthenticatedUser: jest.fn(() =>
    Promise.resolve({ id: 1, username: "authUser", permid: 1 })
  ),
  getUser: jest.fn(() =>
    Promise.resolve({ id: 2, username: "testUser", permid: 2 })
  ),
  updateUserPermId: jest.fn(() => Promise.resolve()),
}));

jest.mock("../../src/service/permission-service", () => ({
  getPermission: jest.fn((id) =>
    Promise.resolve({
      id,
      alterpages: true,
      deletepages: false,
      versions: true,
      allcomments: false,
      tags: true,
      users: false,
    })
  ),
  updatePermissions: jest.fn(() => Promise.resolve()),
  createPermission: jest.fn(() => Promise.resolve(3)),
}));

describe("UserPermissions component tests", () => {
  test("UserPermissions renders correctly", (done) => {
    const wrapper = shallow(<UserPermissions match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });

  test("UserPermissions correctly handles missing permissions", (done) => {
    (userService.getUser as jest.Mock).mockImplementationOnce(() =>
      Promise.resolve({ id: 2, username: "testUser", permid: 0 })
    );

    const wrapper = shallow(<UserPermissions match={{ params: { id: 2 } }} />);

    setTimeout(() => {
      expect(permissionService.createPermission).toHaveBeenCalled();
      expect(userService.updateUserPermId).toHaveBeenCalled();
      done();
    });
  });
});
