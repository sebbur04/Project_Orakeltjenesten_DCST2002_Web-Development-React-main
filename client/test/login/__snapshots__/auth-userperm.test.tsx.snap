// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPermissions component tests UserPermissions renders correctly 1`] = `
<div
  className="mainContent"
>
  <h3>
    Mangler rettigheter
  </h3>
</div>
`;
